server/base/src/main/java/org/apache/accumulo/server/master/state/MetaDataStateStore.java

```diff
@@ -60,136 +64,84 @@ protected MetaDataStateStore(ClientContext context, CurrentState state, String t
   }
 
   @Override
-  public void setLocations(Collection<Assignment> assignments) throws DistributedStoreException {
-    BatchWriter writer = createBatchWriter();
-    try {
-      for (Assignment assignment : assignments) {
-        Mutation m = new Mutation(assignment.tablet.getMetadataEntry());
-        assignment.server.putLocation(m);
-        assignment.server.clearFutureLocation(m);
-        SuspendingTServer.clearSuspension(m);
-        writer.addMutation(m);
-      }
-    } catch (Exception ex) {
-      throw new DistributedStoreException(ex);
-    } finally {
-      try {
-        writer.close();
-      } catch (MutationsRejectedException e) {
-        throw new DistributedStoreException(e);
-      }
-    }
-  }
+  public void setLocations(Assignment assignment, TServerInstance prevLastLoc) {
 
-  BatchWriter createBatchWriter() {
-    try {
-      return context.createBatchWriter(targetTableName,
-          new BatchWriterConfig().setMaxMemory(MAX_MEMORY)
-              .setMaxLatency(LATENCY, TimeUnit.MILLISECONDS).setMaxWriteThreads(THREADS));
-    } catch (Exception e) {
-      throw new RuntimeException(e);
+    TabletMutator tabletMutator = ample.mutateTablet(assignment.tablet);
+    tabletMutator.putLocation(assignment.server, LocationType.CURRENT);
+    tabletMutator.putLocation(assignment.server, LocationType.LAST);
+    tabletMutator.deleteLocation(assignment.server, LocationType.FUTURE);
+
+    // remove the old location
+    if (prevLastLoc != null && !prevLastLoc.equals(assignment.server)) {
+      tabletMutator.deleteLocation(prevLastLoc, LocationType.LAST);
     }
+
+    tabletMutator.mutate();
+
   }
 
   @Override
-  public void setFutureLocations(Collection<Assignment> assignments)
-      throws DistributedStoreException {
-    BatchWriter writer = createBatchWriter();
-    try {
-      for (Assignment assignment : assignments) {
-        Mutation m = new Mutation(assignment.tablet.getMetadataEntry());
-        SuspendingTServer.clearSuspension(m);
-        assignment.server.putFutureLocation(m);
-        writer.addMutation(m);
-      }
-    } catch (Exception ex) {
-      throw new DistributedStoreException(ex);
-    } finally {
-      try {
-        writer.close();
-      } catch (MutationsRejectedException e) {
-        throw new DistributedStoreException(e);
-      }
+  public void setFutureLocations(Collection<Assignment> assignments) {
+
+    for (Assignment assignment : assignments) {
+      TabletMutator tabletMutator = ample.mutateTablet(assignment.tablet);
```

Comment:
A commit I have yet to push to this branch removed the Collection<Assigment> from this function similar to what I did for setLocation. This will remove the for loop. That is still something we want for this function or was that only for setLocation?